web/packages/teleport/src/Discover/SelectResource/types.ts

```diff
@@ -66,10 +66,19 @@ export enum KubeLocation {
   Aws,
 }
 
+/** SAMLServiceProviderPresets defines SAML service provider preset types.
+ * Used to define custom or pre-defined configuration flow.
+*/
+export enum SAMLServiceProviderPresets {
+  Unspecified = 'unspecified',
+  Grafana = 'grafana',
+}
+
 export interface ResourceSpec {
   dbMeta?: { location: DatabaseLocation; engine: DatabaseEngine };
   nodeMeta?: { location: ServerLocation };
   kubeMeta?: { location: KubeLocation };
+  samlMeta?: { preset: SAMLServiceProviderPresets };
```

Comment:
Field name `preset` was chosen as that's what we use to define SAML configuration profile in the backend as well.